File: app/src/main/java/com/paradigmadigital/dagger/di/MainActivityComponent.java
package com.paradigmadigital.dagger.di;

import com.paradigmadigital.dagger.platform.ActivityModule;
import com.paradigmadigital.dagger.platform.MainActivityModule;
import com.paradigmadigital.dagger.ui.main.MainActivity;

import dagger.Subcomponent;
import dagger.android.AndroidInjector;

@PerActivity
@Subcomponent(modules = {ActivityModule.class, MainActivityModule.class})
public interface MainActivityComponent extends AndroidInjector<MainActivity> {

    @Subcomponent.Builder
    abstract class Builder extends AndroidInjector.Builder<MainActivity> {
    }
}

File: module_main/src/main/java/eu/sesma/dagger/main/di/MainActivityComponent.kt
package eu.sesma.dagger.main.di

import android.support.v7.app.AppCompatActivity
import dagger.Component
import eu.sesma.dagger.core.di.ActivityScoped
import eu.sesma.dagger.core.di.CoreComponent
import eu.sesma.dagger.main.MainActivity


@ActivityScoped
@Component(
        dependencies = [CoreComponent::class],
        modules = [MainActivityModule::class]
)
interface MainActivityComponent {

    fun inject(activity: MainActivity)

    //Exposed to sub-graphs.
    fun activity(): AppCompatActivity
}